compiler/resolution/functionResolution.cpp

```diff
@@ -5745,37 +5742,13 @@ static FnSymbol* adjustAndResolveForwardedCall(CallExpr* call, ForwardingStmt* d
   return ret;
 }
 
-static void detectForwardingCycle(CallExpr* call) {
-  BlockStmt* cur = toBlockStmt(call->getStmtExpr()->parentExpr);
-  DefExpr* firstDef = NULL;
-  while (cur != NULL) {
-    DefExpr* def = toDefExpr(cur->body.head);
-    if (def == NULL || def->sym->name != astr_chpl_forward_tgt)
-      return; // not a cycle
-
-    if (firstDef == NULL) {
-      firstDef = def;
-      INT_ASSERT(firstDef->sym->type && firstDef->sym->type != dtUnknown);
-    } else {
-      // If firstDef has same type as def, cycle is found.
-      if (firstDef->sym->type == def->sym->type) {
-        Type* t = canonicalDecoratedClassType(firstDef->sym->getValType());
-        TypeSymbol* ts = t->symbol;
-        USR_FATAL_CONT(def, "forwarding cycle detected");
-        USR_PRINT(ts, "for the type %s", ts->name);
-        USR_STOP();
-      }
-    }
-    cur = toBlockStmt(cur->parentExpr);
-  }
-}
-
+llvm::SmallVector<std::tuple<AggregateType*, const char*, const char*>, 4> forwardCallCycleSet;
```

Comment:
Normally I'd be a bit more concerned with adding a global like this, but I think dyno will let us remove this chunk of code eventually so I'm alright with it for now.

Reply:
I think it's alright as-is, since testing is passing.